### PR TITLE
Fixes category search functionality for vector layers

### LINKED ISSUE #1085: [Category widget] Fix search
Search is not working when using vector layers:
- It shows "Your search "" didn't match with any value." initially.
- Typing in the search box tries to fetch data from http://host/search?q=something

Summary always show "SEARCH IN 6 CATEGORIES" instead of showing the total number of categories.

### LINKED ISSUE #1086: [Category widget] Fix totals in the summary
Summary always shows -> 0% OF TOTAL.

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
Fixes #1085 and #1086 (two for the price of one):

search

@xavijam can you PTAL! Thanks!

### CODE REVIEW

File: src/dataviews/category-dataview-model.js [hunk 1]
```diff
@@ -152,7 +151,7 @@ module.exports = DataviewModelBase.extend({
   },
 
   getCount: function () {
-    return this.get('allCategoryNames').length;
+    return this.get('categoriesCount');
```
Comment (xavijam) [inline src/dataviews/category-dataview-model.js]: Why this change?

File: src/dataviews/category-dataview-model.js [hunk 2]
```diff
@@ -89,6 +80,14 @@ module.exports = DataviewModelBase.extend({
     this._searchModel.bind('change:data', this._onSearchDataChange, this);
   },
 
+  _onSearchDataChange: function () {
```
Comment (xavijam) [inline src/dataviews/category-dataview-model.js]: I feel I've already done that some days ago, haven't I?

Comment (xavijam) [inline src/dataviews/category-dataview-model.js]: Ahh, you have moved it :+1:

**Comment by xavijam** [comment]
:+1:

File: src/dataviews/category-dataview-model.js (hunk 1 above)
Comment (alonsogarciapablo) [inline src/dataviews/category-dataview-model.js]: That's we had originally: https://github.com/CartoDB/cartodb.js/blob/v4/src/dataviews/category-dataview-model.js#L155

Comment (xavijam) [inline src/dataviews/category-dataview-model.js]: And why this change says that it was using 'allCategoryNames' size?
I'm confused.

**Pull request merged**